weave-js/src/components/PagePanelComponents/Home/Browse3/components/TraceNavigator/TraceViews/TreeView.tsx

```diff
@@ -0,0 +1,599 @@
+            <div className="text-right">
+              {showDuration
+                ? duration !== null
+                  ? formatDuration(duration)
+                  : ''
+                : cost && (
+                    <TraceStat
+                      label={cost}
+                      tooltip={
+                        <div className="text-white-800">
+                          {costToolTipContent}
+                          {tokens && (
+                            <>
+                              <br />
+                              <span style={{fontWeight: 600}}>
+                                Estimated tokens
+                              </span>
+                            </>
+                          )}
+                          {tokens && tokenToolTipContent}
+                        </div>
+                      }
+                      className="text-xs text-moon-400"
+                    />
+                  )}
+            </div>
+
+            <Icon
+              name={getCallTypeIcon(typeName)}
+              className={`max-w-16 max-h-16 ${opTypeColor}`}
+            />
+            <StatusChip value={statusCode} iconOnly />
+          </div>
+        </div>
+      </Button>
+    </div>
+  );
+};
+
+interface TreeViewHeaderProps {
+  searchQuery: string;
+  onSearchChange: (value: string) => void;
+  showDuration: boolean;
+  onToggleView: () => void;
+  strictSearch: boolean;
+  onToggleStrictSearch: () => void;
+}
+
+const TreeViewHeader: React.FC<TreeViewHeaderProps> = ({
+  searchQuery,
+  onSearchChange,
+  showDuration,
+  onToggleView,
+  strictSearch,
+  onToggleStrictSearch,
+}) => {
+  return (
+    <div className="flex items-center px-8 py-4 text-sm">
+      <TextField
+        value={searchQuery}
+        onChange={onSearchChange}
+        placeholder="Filter by op name..."
+        icon="filter-alt"
+        extraActions={
+          searchQuery !== '' && (
+            <>
+              <Button
+                variant="ghost"
+                size="small"
+                onClick={onToggleStrictSearch}
+                tooltip={
+                  strictSearch
+                    ? '(Strict) Only show matching nodes'
+                    : '(Loose) Show children of matching nodes'
+                }
+                className={strictSearch ? 'mr-2' : 'mr-2 text-moon-500'}
+                active={strictSearch}>
+                Strict
+              </Button>
+              <Button
+                variant="ghost"
+                size="small"
+                icon="close"
+                onClick={() => onSearchChange('')}
+                className="mr-6"
+              />
+            </>
+          )
+        }
+      />
+      <Button
+        variant="ghost"
+        icon={showDuration ? 'recent-clock' : 'database-artifacts'}
+        onClick={onToggleView}
+        className="ml-8"
+        tooltip={showDuration ? 'Latency' : 'Cost / duration'}
```

Comment:
The icon and tooltip on this button is pretty confusing

Reply:
fixed